api/envoy/extensions/dynamic_modules/v3/dynamic_modules.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+
+package envoy.extensions.dynamic_modules.v3;
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.dynamic_modules.v3";
+option java_outer_classname = "DynamicModulesProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/dynamic_modules/v3;dynamic_modulesv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: Dynamic Modules common configuration]
+
+// Configuration of a dynamic module. A dynamic module is a shared object file that can be loaded via dlopen
+// by a various Envoy extension points. Currently, only HTTP filter (envoy.filters.http.dynamic_modules) is supported.
+//
+// How the module is loaded is determined by the extension point that uses it. For example, the HTTP filter
+// loads the module with dlopen on the first creation of the filter chain for the same shared object file.
+//
+// Whether or not the shared object is the same is determined by the file path as well as the file's inode depending
+// on the platform. Notably, if the file path and the content of the file are the same, the shared object will be reused.
+//
+// A module must be compatible with the ABI specified in :repo:`abi.h <source/extensions/dynamic_modules/abi.h>`.
+// Currently, compatibility is only guaranteed by an exact version match between the Envoy
+// codebase and the dynamic module SDKs. In the future, after the ABI is stabilized, we will revisit
+// this restriction and hopefully provide a wider compatibility guarantee. Until then, Envoy
+// checks the hash of the ABI header files to ensure that the dynamic modules are built against the
+// same version of the ABI.
+//
+// Currently, the implementation is work in progress and not usable.
+message DynamicModuleConfig {
+  // The name of the dynamic module. The client is expected to have some configuration indicating where to search for the module.
+  // In Envoy, the search path can only be configured via the environment variable ``ENVOY_DYNAMIC_MODULES_SEARCH_PATH``.
+  // The actual search path is ``${ENVOY_DYNAMIC_MODULES_SEARCH_PATH}/lib${name}.so``. TODO: make the search path configurable via
+  // command line options.
+  string name = 1 [(validate.rules).string = {min_len: 1}];
+
+  // Set true to prevent the module from being unloaded with dlclose.
+  // This is useful for modules that have global state that should not be unloaded.
+  bool do_not_close = 3;
```

Comment:
When would it be unloaded? Like when the configuration is unloaded? Can you clarify?

Reply:
I think we are now inclined to load all SOs at the startup time, so probably I am deleting this option in the first place

Reply:
https://github.com/envoyproxy/envoy/pull/36448/commits/487cca7e10cc8d53d54aab14b111e0ae6e957b1b